pyomo/solvers/tests/checks/test_BARON.py

```diff
@@ -0,0 +1,32 @@
+"""Tests the BARON interface."""
+import pyutilib.th as unittest
+from pyomo.environ import (ConcreteModel, Constraint, Objective, Var, log10,
+                           minimize)
+from pyomo.opt import SolverFactory, TerminationCondition
+
+# I don't know how you figure out if BARON is available
+baron_available = True

```

Comment:
On the Baron and Gurobi plugins, in particular, it's also necessary to check if the licenses are valid. You should call both the `available()` and `license_is_valid()` methods. Another approach, that I use in many places, is to piggyback off of the list of solver test cases in `pyomo.solvers.tests.solvers`, where all of this logic is already implemented and the result is cached onto the `available` property on each test case. E.g.,

```python
from pyomo.solvers.tests.solvers import test_solver_cases
baron_available = test_solver_cases('baron', 'bar').available
```